### ISSUE TITLE
Add donation popup

### ISSUE THREAD

**Issue by Popolechien** [kiwix/kiwix-android#3736]
[Signal](../../../../signalapp) has a donation prompt that appears every three months or so at the bottom of its android app. Nothing fancy (see text below - it's in French but the gist of it is `Donate today / Signal needs your help` with the choice button being _remind me later_ of _donate_)

There is already a static feature in the settings, so this is in addition to the latter.

Signal has a sophisticated donation funnel that allows in-app payments, but at this stage a link to support.kiwix.org would already be good

![donation_prompt](https://github.com/kiwix/kiwix-android/assets/21675885/78f71b50-06cb-467f-b1e6-8329104be23b)

**Comment by kelson42** [comment]
@MohitMaliDeveloper Please implement something like in Signal.

**Comment by kelson42** [comment]
The donation does not look well on tablet in dark mode, please make abetter proposal.
![Screenshot_20240919_195245_Kiwix](https://github.com/user-attachments/assets/75eba12f-dc70-4b15-927a-19bbbed73783)

**Comment by MohitMaliFtechiz** [comment]
@kelson42 For tablets we can show this in the middle width half of the screen size width, and when the tablet is running in portrait mode we can show the the donation dialog something like this. @kelson42 What do you think?


| (Device width/ 1.5)  | (Device width/ 1.2) |
| ------------- | ------------- |
| ![Screenshot from 2024-09-20 17-27-27](https://github.com/user-attachments/assets/20c556ce-0643-44d0-b620-90c63f644c39)  | ![Screenshot from 2024-09-20 17-32-18](https://github.com/user-attachments/assets/57a00283-ed57-431a-ac1b-7765dd69824b)  |





![Screenshot from 2024-09-20 17-17-59](https://github.com/user-attachments/assets/3e496e54-95e7-41ad-ad3b-34c18145e34a)

**Comment by kelson42** [comment]
@MohitMaliFtechiz Can you please confirm this works perfectly for custom apps?

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz Can you please confirm this works perfectly for custom apps?

@kelson42 Yes, I can confirm it is working fine with custom apps. The donation popup will only show when the `support_url` is configured for the custom apps(And clicking on the donation button will open the [support URL](https://github.com/kiwix/kiwix-android-custom/blob/bada9cf7b1445dbedeabbe14c6ae7d4ed0d1a7a0/tunisie/info.json#L5C4-L5C15)).


| When no support URL is configured  | When support URL is configured |
| ------------- | ------------- |
| ![Image](https://github.com/user-attachments/assets/e051c749-c032-4842-aac1-31f7360a3fda) | ![Image](https://github.com/user-attachments/assets/eab109c9-fda2-41c7-a2df-ab9cef37bd87) |

**Comment by MohitMaliFtechiz** [comment]
> I reopen the issue because I feel spamed by the donation box:
> it appears quickly after app installation and it should not
> If I click later (which is the only choice beside paying) it comes back soon in a few days
> IMHO it should appears only 3 months after install and the once every three months if later is click ked.

@kelson42 It will show again in 3 days if the user clicks on the "Later" button, we have documented this in point number 2 of our PR description(https://github.com/kiwix/kiwix-android/pull/3984#issue-2507838055). But your point is valid, I am making the changes.
